### PR TITLE
Preserve return attribute

### PR DESCRIPTION
**Pull request by kontomondo**
addresses #800

### CODE REVIEW

File: src/Fantomas.Tests/LetBindingTests.fs [hunk 1]
```diff
@@ -546,4 +546,10 @@ let ``multiple empty lines between equals and expression`` () =
...
 
     ()
-"""
\ No newline at end of file
+"""
+
+[<Test>]
+let ``should preserve return attributes`` () =
+    formatSourceString false """let f x : [<return: MyCustomAttributeThatWorksOnReturns>] int = x""" config
+    |> should equal """let f x: [<return: MyCustomAttributeThatWorksOnReturns>] int = x
```
Comment (kontomondo) [inline src/Fantomas.Tests/LetBindingTests.fs:554]: this is failing at the moment with:
```
  Expected string length 65 but was 68. Strings differ at index 11.
  Expected: "...: [<return: MyCustomAttributeThatWorksOnReturns>] int = x\n"
  But was:  "...: [<``return``:MyCustomAttributeThatWorksOnReturns>] int = x\n"
  ------------------^
```

Comment (nojaf) [reply src/Fantomas.Tests/LetBindingTests.fs:554]: I would replace the active pattern in SourceParser.fs to
```fsharp
let (|Attribute|) (a : SynAttribute) =
    let (LongIdentWithDots s) = a.TypeName
    (s, a.ArgExpr, Option.map (fun (i:Ident) -> i.idText) a.Target)
```

File: src/Fantomas.Tests/LetBindingTests.fs [hunk 2]
```diff
@@ -546,4 +546,10 @@ let ``multiple empty lines between equals and expression`` () =
 
 
     ()
-"""
\ No newline at end of file
+"""
+
+[<Test>]
```
Comment (nojaf) [inline src/Fantomas.Tests/LetBindingTests.fs:551]: Makes more sense to have the test in AttributeTests.fs

File: src/Fantomas.Tests/LetBindingTests.fs (hunk 1 above)
Comment (nojaf) [inline src/Fantomas.Tests/LetBindingTests.fs:554]: No space between `:`, see ` ``should handle targets on attributes`` ` test

File: src/Fantomas.Tests/LetBindingTests.fs (hunk 2 above)
Comment (kontomondo) [reply src/Fantomas.Tests/LetBindingTests.fs]: done

File: src/Fantomas.Tests/LetBindingTests.fs (hunk 1 above)
Comment (kontomondo) [reply src/Fantomas.Tests/LetBindingTests.fs]: done

Comment (kontomondo) [reply src/Fantomas.Tests/LetBindingTests.fs]: done

File: src/Fantomas/CodePrinter.fs [hunk 3]
```diff
@@ -450,7 +450,7 @@ and addSpaceAfterGenericConstructBeforeColon ctx =
         sepNone
     <| ctx
 
-and genExprSepEqPrependType astContext (pat:SynPat) (e: SynExpr) (isPrefixMultiline: bool) ctx =
+and genExprSepEqPrependType astContext (pat:SynPat) (e: SynExpr) (valInfo:SynValInfo option) (isPrefixMultiline: bool) ctx =
```
Comment (kontomondo) [inline src/Fantomas/CodePrinter.fs:453]: should the order of this parameter be changed or perhaps name changed to be more in line with existing code ?

**Comment by kontomondo** [comment]
@nojaf is there anything else needed for this to get merged ?

File: src/Fantomas/CodePrinter.fs [hunk 4]
```diff
@@ -489,7 +489,7 @@ and genExprSepEqPrependType astContext (pat:SynPat) (e: SynExpr) (valInfo:SynVal
         let genMetadataAttributes ctx =
```
[the remark this answers is not in the document]
Comment (kontomondo) [reply src/Fantomas/CodePrinter.fs]: absolutely, done

**Comment by nojaf** [comment]
Many thanks @kontomondo! Sorry for the delay.

**Comment by kontomondo** [comment]
@nojaf Thank you, and no worries, that was blazingly fast by open source standards :)

**Pull request merged**